pkg/ingest/store/factory.go

```diff
@@ -0,0 +1,112 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+)
+
+var (
+	ErrNoStorageAdapter = errors.New("no storage adapter found")
+)
+
+type ObjectStoreEntry struct {
+	// FullKey represents the fully qualified path in the object store namespace for the given entry
+	FullKey string
+	// RelativeKey represents a path relative to prefix (or directory). If none specified, will be identical to FullKey
+	RelativeKey string
```

Comment:
It is used by the marker as the last key and passed as-is to the client. The client should pass it to the server in the next call. It is important to keep the full key in the object store and not leave it to the client it to parse it from the combination of `RelativeKey` & `StorageURI` because of `/` complexities and the way different blockstores work.